# Hello you, 👋
### and welcome to this assignment. Let's dive right in!

If you haven't read [the README](https://github.com/Kanyelings/projects-3.10/blob/main/README.md) for this project, make sure to do so immediately.

This notebook is divided into two sections for now: **Wrangling** and **Analyzing** (except otherwise in the future).
Feel free to send your questions to [this mail](mailto:d3liaayoko).

Each empty cell in this notebook should contain the solution to the task above it. Also, please do not clear your outputs!

## 1. Wrangling

**Task 1.1:** Import the dataset`("winequality-red.csv")` and display the first five rows of the DataFrame.

In [2]:
import pandas as pd

# Import the CSV file and read it into a DataFrame
df = pd.read_csv("wine-quality dataset/winequality-red.csv")

# Display the first five rows of the DataFrame
print(df.head())

   fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
0            7.4              0.70         0.00             1.9      0.076   
1            7.8              0.88         0.00             2.6      0.098   
2            7.8              0.76         0.04             2.3      0.092   
3           11.2              0.28         0.56             1.9      0.075   
4            7.4              0.70         0.00             1.9      0.076   

   free sulfur dioxide  total sulfur dioxide  density    pH  sulphates  \
0                 11.0                  34.0   0.9978  3.51       0.56   
1                 25.0                  67.0   0.9968  3.20       0.68   
2                 15.0                  54.0   0.9970  3.26       0.65   
3                 17.0                  60.0   0.9980  3.16       0.58   
4                 11.0                  34.0   0.9978  3.51       0.56   

   alcohol  quality  
0      9.4        5  
1      9.8        5  
2      9.8        5 

**Task 1.2:** Inspect the DataFrame using [any 3 of these ways.](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&cad=rja&uact=8&ved=2ahUKEwi-gZHst4uAAxWLRaQEHf_GA_4QFnoECA8QAQ&url=https%3A%2F%2Fmedium.com%2Fcodex%2F10-efficient-ways-for-inspecting-a-pandas-dataframe-object-3f66563e2f2&usg=AOvVaw2XnUUADHfXp3llJVL3JClr&opi=89978449). If there are any missing values, drop them.

In [9]:
# inspect the DataFrame for missing values
print("inspect the DataFrame for missing values")
print(df.isnull())


# drop the missing values and assign the cleaned DataFrame back to df
df = df.dropna()

# inspect the cleaned DataFrame for missing values
print("inspect the cleaned DataFrame for missing values")
print(df.isnull())


inspect the DataFrame for missing values
      fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
0             False             False        False           False      False   
1             False             False        False           False      False   
2             False             False        False           False      False   
3             False             False        False           False      False   
4             False             False        False           False      False   
...             ...               ...          ...             ...        ...   
1594          False             False        False           False      False   
1595          False             False        False           False      False   
1596          False             False        False           False      False   
1597          False             False        False           False      False   
1598          False             False        False           False  

**Task 1.3:** Drop the `quality` column.

In [11]:
# drop the 'Quality' column and assign the modified DataFrame back to df
df = df.drop('quality', axis=1)

# print the modified DataFrame
print(df)

      fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
0               7.4             0.700         0.00             1.9      0.076   
1               7.8             0.880         0.00             2.6      0.098   
2               7.8             0.760         0.04             2.3      0.092   
3              11.2             0.280         0.56             1.9      0.075   
4               7.4             0.700         0.00             1.9      0.076   
...             ...               ...          ...             ...        ...   
1594            6.2             0.600         0.08             2.0      0.090   
1595            5.9             0.550         0.10             2.2      0.062   
1596            6.3             0.510         0.13             2.3      0.076   
1597            5.9             0.645         0.12             2.0      0.075   
1598            6.0             0.310         0.47             3.6      0.067   

      free sulfur dioxide  

**Task 1.4:** Subset the DataFrame such that it masks only wine with `alcohol` value greater than or equal to **12.0**.

In [12]:
# subset the DataFrame based on the condition "alcohol >= 12.0"
df_subset = df[df['alcohol'] >= 12.0]

# print the subsetted DataFrame
print(df_subset)

      fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
45              4.6              0.52         0.15             2.1      0.054   
95              4.7              0.60         0.17             2.3      0.058   
131             5.6              0.50         0.09             2.3      0.049   
132             5.6              0.50         0.09             2.3      0.049   
142             5.2              0.34         0.00             1.8      0.050   
...             ...               ...          ...             ...        ...   
1538            6.2              0.64         0.09             2.5      0.081   
1547            6.3              0.60         0.10             1.6      0.048   
1570            6.4              0.36         0.53             2.2      0.230   
1573            6.0              0.58         0.20             2.4      0.075   
1588            7.2              0.66         0.33             2.5      0.068   

      free sulfur dioxide  

**Task 1.5:** Subset the DataFrame again, this time, masking only wine with `pH` value greater than or equal to **2.0**.

In [13]:
# subset the DataFrame based on the condition "pH >= 2.0"
df_subset = df[df['pH'] >= 2.0]

# print the subsetted DataFrame
print(df_subset)

      fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
0               7.4             0.700         0.00             1.9      0.076   
1               7.8             0.880         0.00             2.6      0.098   
2               7.8             0.760         0.04             2.3      0.092   
3              11.2             0.280         0.56             1.9      0.075   
4               7.4             0.700         0.00             1.9      0.076   
...             ...               ...          ...             ...        ...   
1594            6.2             0.600         0.08             2.0      0.090   
1595            5.9             0.550         0.10             2.2      0.062   
1596            6.3             0.510         0.13             2.3      0.076   
1597            5.9             0.645         0.12             2.0      0.075   
1598            6.0             0.310         0.47             3.6      0.067   

      free sulfur dioxide  

## 2. Analyzing

For the next series of tasks. The instructions will be updated as soon as the others have been completed.

Have fun, champ!

**Task 2.1:**

**Task 2.2:**

**Task 2.3:**